diff --git a/qsp/game.c b/qsp/game.c
index d33c936..d99ee48 100644
--- a/qsp/game.c
+++ b/qsp/game.c
@@ -243,7 +243,7 @@ void qspOpenQuestFromData(char *data, int dataSize, QSP_CHAR *fileName, QSP_BOOL
 	if (!qspCheckQuest(strs, count, isUCS2))
 	{
 		qspSetError(QSP_ERR_CANTLOADFILE);
-		qspFreeStrs(strs, count);
+		qspFreeStrs((void **)strs, count);
 		return;
 	}
 	buf = qspGameToQSPString(strs[0], isUCS2, QSP_FALSE);
@@ -308,7 +308,7 @@ void qspOpenQuestFromData(char *data, int dataSize, QSP_CHAR *fileName, QSP_BOOL
 		else
 			ind += actsCount * (isOldFormat ? 2 : 3);
 	}
-	qspFreeStrs(strs, count);
+	qspFreeStrs((void **)strs, count);
 	qspLocsCount = end;
 	qspCreateWorld(end, locsCount);
 	count = locsCount - start;
@@ -528,7 +528,7 @@ void qspOpenGameStatusFromString(QSP_CHAR *str)
 	if (!qspCheckGameStatus(strs, count))
 	{
 		qspSetError(QSP_ERR_CANTLOADFILE);
-		qspFreeStrs(strs, count);
+		qspFreeStrs((void **)strs, count);
 		return;
 	}
 	++qspRefreshCount;
@@ -627,7 +627,7 @@ void qspOpenGameStatusFromString(QSP_CHAR *str)
 			}
 		}
 	}
-	qspFreeStrs(strs, count);
+	qspFreeStrs((void **)strs, count);
 	qspIsMainDescChanged = qspIsVarsDescChanged = qspIsObjectsChanged = qspIsActionsChanged = QSP_TRUE;
 	qspOpenIncludes();
 	qspCurLoc = qspLocIndex(locName);
